P/PMIx/build_tarballs.jl

```diff
@@ -33,6 +33,9 @@ make install
 # platforms are passed in on the command line
 platforms = supported_platforms()
 
+# PMIx does not support 32-bit builds <https://docs.openpmix.org/en/latest/release-notes/platform.html>
+filter!(p -> nbits(p) != 32, platforms)
```

Comment:
This makes you wonder how it worked before, but maybe it was an accident?  Anyway, I see this is only used as dependency of OpenMPI@v5 and prrte (which is also a dependency of OpenMPI@v5), so limiting to 32-bit platforms shouldn't be a problem, since OpenMPI@v5 is 64-bit only too.